add_gist.rb

```diff
@@ -0,0 +1,62 @@
+require 'rubygems'
+require 'net/http'
+require 'uri'
+require 'json'
+require 'OpenSSL'
+require 'find'
+require 'dotenv/load'
+
+# Module that gathers the methods necessary to add a new gist
+module AddGist
+  GISTS_URL = 'https://api.github.com/gists'.freeze
+
+  module_function
+
+  def read_files(pwd = Dir.pwd)
+    files = {}
+    Find.find(pwd) do |path|
+      files.merge!(File.basename(path) => { 'content': File.open(path, 'r+').read }) unless File.directory?(path)
+    end
+    files
+  rescue Errno::ENOENT => e
+    puts e.message
+    exit
+  end
+
+  def send_request(files, description, is_public, token)
+    uri = URI.parse(GISTS_URL)
+    uri.query = URI.encode_www_form(access_token: token)
+
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = true
+    http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+
+    request = Net::HTTP::Post.new(uri.request_uri)
+    request.body = {
+      "description": description.to_s,
+      "public": is_public,
+      "files": files
+    }.to_json
+
+    http.request(request)
+  end
+
+  if ARGV.length != 3
+    puts %(Usage: ruby addGist.rb <dirname/filename> <public? (boolean)> <"gist description">)
+    exit
+  end
+  path = ARGV[0]
+  is_public = ARGV[1].casecmp?('true')
+  gist_description = ARGV[2]
+
+  files = read_files(path)
+  token = ENV['ACCESS_TOKEN']
+  response = send_request(files, gist_description, is_public, token)
+
+  if response.code == '201'
+    html_url = JSON.parse(response.body)['html_url']
+    puts "Gist created successfully! \nAccess URL: #{html_url}"
+  else
+    puts response.code
+  end
```

Comment:
I am still not convinced with having all this lying around. Maybe have it in a method and call it at the end?

Reply:
that way we can make sensitive methods private